Description: update to mkdirp ≥ 1
Author: Yadd <yadd@debian.org>
Forwarded: https://github.com/davglass/cpr/issues/68
Last-Update: 2020-10-21

--- a/lib/index.js
+++ b/lib/index.js
@@ -121,12 +121,13 @@
                     err.errno = 27;
                     options.errors.push(err);
                 } else {
-                    mkdirp(to, stat.mode, stack.add(function(err) {
+                    var ef = stack.add(function(err) {
                         /*istanbul ignore next*/
                         if (err) {
                             options.errors.push(err);
                         }
-                    }));
+                    });
+                    mkdirp(to, stat.mode).then(ef).catch(ef);
                 }
             }));
         }
@@ -139,7 +140,7 @@
 
 var copyFile = function(from, to, options, callback) {
     var dir = path.dirname(to);
-    mkdirp(dir, function() {
+    mkdirp(dir).then( () => {
         fs.stat(to, function(statError) {
             var err;
             if(!statError && options.overwrite !== true) {
